src/umpire/resource/HostMpi3SharedMemoryResource.cpp

```diff
@@ -0,0 +1,66 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-24, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+
+#include "umpire/resource/HostMpi3SharedMemoryResource.hpp"
+
+#include "umpire/resource/MemoryResource.hpp"
+#include "umpire/util/Macros.hpp"
+#include "umpire/util/error.hpp"
+
+namespace umpire {
+namespace resource {
+
+HostMpi3SharedMemoryResource::HostMpi3SharedMemoryResource(const std::string& name, int id, MemoryResourceTraits traits)
+    : MemoryResource{name, id, traits}
+{
+  constexpr int IGNORE_KEY{0};
+  MPI_Comm_split_type(MPI_COMM_WORLD, MPI_COMM_TYPE_SHARED, IGNORE_KEY, MPI_INFO_NULL, &m_shared_comm);
+  MPI_Comm_rank(m_shared_comm, &m_local_rank);
+}
+
+HostMpi3SharedMemoryResource::~HostMpi3SharedMemoryResource()
+{
+  MPI_Comm_free(&m_shared_comm);
+}
+
+void* HostMpi3SharedMemoryResource::allocate(std::size_t bytes)
+{
+  void* ptr;
+  MPI_Win win;
+  MPI_Aint size = (m_local_rank != 0) ? 0 : bytes;
```

Comment:
Just to avoid a double negative, I think this would be easier understood as

`MPI_Aint size = (m_local_rank == 0) ? bytes : 0;